webapi/gtkui.py

```diff
@@ -13,7 +19,96 @@ class GtkUI(GtkPluginBase):
     def enable(self):
         """Triggers when plugin is enabled."""
         LOGGER.info('Enabling WebAPI plugin GTK ...')
+        self.glade = gtk.glade.XML(get_resource("config.glade"))
+
+        component.get("Preferences").add_page("WebAPI", self.glade.get_widget("prefs_box"))
+        component.get("PluginManager").register_hook("on_apply_prefs", self.on_apply_prefs)
+        component.get("PluginManager").register_hook("on_show_prefs", self.on_show_prefs)
+
+        self.glade.get_widget('add_button').connect("clicked", self.add_domain)
+        self.glade.get_widget('remove_button').connect("clicked", self.remove_domain)
+        self.glade.get_widget('remove_all_button').connect("clicked", self.remove_all_domain)
+        self.create_listview()
 
     def disable(self):
         """Triggers when plugin is disabled."""
         LOGGER.info('Disabling WebAPI plugin GTK ...')
+
+        component.get("Preferences").remove_page("WebAPI")
+        component.get("PluginManager").deregister_hook("on_apply_prefs", self.on_apply_prefs)
+        component.get("PluginManager").deregister_hook("on_show_prefs", self.on_show_prefs)
+
+    def on_apply_prefs(self):
+        log.debug("applying prefs for WebAPI")
+        config = {}
+        if self.glade.get_widget("enable_cors").get_active():
+            config["enable_cors"] = True
+        else:
+            config["enable_cors"] = False
+
+        config["allowed_origin"] = []
+        if len(self.model) != 0:
+            # remove all the entries in the model
+            for i in range(len(self.model)):
+                iter = self.model.get_iter(i)
```

Comment:
Here and elsewhere `iter` is a built-in function, and we'd rather not to shadow it.